### PR TITLE
Group 4

### PR DESCRIPTION
**Pull request by PtrBld**

### CODE REVIEW

File: library.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
```
Comment (sajid3900) [inline library.py:8]: Class description should be in PEP8 style. Use ''' ''' to describe the class.

File: library.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    def print_longest_crime_audio_book(self) -> str:
```
Comment (maxftl) [inline library.py:9]: Function description is not matching method name (does it print all audio books, or just the longest). Why does the function return something?

File: library.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        crime_audio_books: list[Book] = []
+        max: Optional[Book] = None
+
+        # print banner
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
```
Comment (sajid3900) [inline library.py:18]: use better name for the variable. 'book' would be better as it is more readable

File: library.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
+    def print_longest_crime_audio_book(self) -> str:
+        crime_audio_books: list[Book] = []
+        max: Optional[Book] = None
```
Comment (maxftl) [inline library.py:11]: please change the variable name to something more descriptive that doesn't conflict with built in function names

File: library.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
+                    if max is not None and max.duration < b.duration:
+                        max = b
+                    # end if
+                # end if
+            else:
```
Comment (sajid3900) [inline library.py:25]: Maybe max has to be set to b.duration here

File: library.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+                pass
+            # end else
+        # end for
+
+        print(f"Title: {max.title}")
```
Comment (maxftl) [inline library.py:30]: Check if max is None

File: library.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        max: Optional[Book] = None
+
+        # print banner
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
```
Comment (sajid3900) [inline library.py:20]: avoid using private variables. Maybe implement getBookType() in Book.py

**Review by maxftl — CHANGES_REQUESTED** [review]
Plz change

**Pull request closed without merging**